meteor/server/api/translationsBundles.ts

```diff
@@ -0,0 +1,88 @@
+import { Meteor } from 'meteor/meteor'
+import {
+	TranslationsBundles as TranslationsBundleCollection,
+	TranslationsBundleId,
+	Translation,
+	TranslationsBundle as DBTranslationsBundle,
+} from '../../lib/collections/TranslationsBundles'
+import {
+	I18NextData,
+	TranslationsBundle as BlueprintTranslationsbundle,
+	TranslationsBundleType,
+} from '@sofie-automation/blueprints-integration'
+import { getHash, protectString, unprotectString } from '../../lib/lib'
+import { BlueprintId } from '../../lib/collections/Blueprints'
+import { Mongocursor } from '../../lib/typings/meteor'
+
+/**
+ * Insert or update translation bundles in the database.
+ *
+ * @param bundles the bundles to insert or update
+ * @param originBlueprintId id of the blueprint the translation bundles belongs to
+ */
+export function upsertBundles(bundles: BlueprintTranslationsbundle[], originBlueprintId: BlueprintId) {
+	for (const bundle of bundles) {
+		const { type, language, data } = bundle
+
+		if (type !== TranslationsBundleType.I18NEXT) {
+			throw new Error(`Unknown bundle type ${type}`)
+		}
+
+		const _id = getExistingId(originBlueprintId, language) || createBundleId(originBlueprintId, language)
```

Comment:
Question: Is there a specific reason for the lookup with `getExistingId`? Why not just use createBundleId right away? (that would be more performant). If there is a reason, plz add it as a comment in the code.